packages/flutter/lib/src/rendering/semantics.dart

```diff
@@ -685,20 +700,47 @@ class SemanticsNode extends AbstractNode {
   }
 
   /// Returns a string representation of this node and its descendants.
-  String toStringDeep([String prefixLineOne = '', String prefixOtherLines = '']) {
+  ///
+  /// The order in which the children of the [SemanticsNode] will be printed can
+  /// be controlled with the [childOrder] parameter.
+  String toStringDeep(DebugSemanticsDumpOrder childOrder, [
+    String prefixLineOne = '',
+    String prefixOtherLines = ''
+  ]) {
+    assert(childOrder != null);
     final StringBuffer result = new StringBuffer()
       ..write(prefixLineOne)
       ..write(this)
       ..write('\n');
     if (_children != null && _children.isNotEmpty) {
-      for (int index = 0; index < _children.length - 1; index += 1) {
-        final SemanticsNode child = _children[index];
-        result.write(child.toStringDeep("$prefixOtherLines \u251C", "$prefixOtherLines \u2502"));
+      final List<SemanticsNode> childrenInOrder = _getChildrenInOrder(childOrder);
+      for (int index = 0; index < childrenInOrder.length - 1; index += 1) {
+        final SemanticsNode child = childrenInOrder[index];
+        result.write(child.toStringDeep(childOrder, "$prefixOtherLines \u251C", "$prefixOtherLines \u2502"));
       }
-      result.write(_children.last.toStringDeep("$prefixOtherLines \u2514", "$prefixOtherLines  "));
+      result.write(childrenInOrder.last.toStringDeep(childOrder, "$prefixOtherLines \u2514", "$prefixOtherLines  "));
     }
     return result.toString();
   }
+
+  Iterable<SemanticsNode> _getChildrenInOrder(DebugSemanticsDumpOrder childOrder) {
+    assert(childOrder != null);
+    switch(childOrder) {
+      case DebugSemanticsDumpOrder.traversal:
+        return new List<SemanticsNode>.from(_children)..sort(_nodeComparator);
+      case DebugSemanticsDumpOrder.inverseHitTest:
+        return _children;
+    }
+    assert(false);
+    return null;
+  }
+
+  static int _nodeComparator(SemanticsNode a, SemanticsNode b) {
```

Comment:
maybe rename this to _geometryComparator or some such to make it obvious _how_ it compares the nodes (that it compares nodes is obvious from the signature!)